homeassistant/components/homeassistant_hardware/firmware_config_flow.py

```diff
@@ -150,98 +137,72 @@ async def _probe_firmware_info(
-                description_placeholders=self._get_translation_placeholders(),
-            )
-
-        # Allow the stick to be used with ZHA without flashing
-        if (
-            self._probed_firmware_info is not None
-            and self._probed_firmware_info.firmware_type == ApplicationType.EZSP
-        ):
-            return await self.async_step_confirm_zigbee()
-
-        if not is_hassio(self.hass):
-            return self.async_abort(
-                reason="not_hassio",
-                description_placeholders=self._get_translation_placeholders(),
-            )
-
-        # Only flash new firmware if we need to
-        fw_flasher_manager = get_zigbee_flasher_addon_manager(self.hass)
-        addon_info = await self._async_get_addon_info(fw_flasher_manager)
-
-        if addon_info.state == AddonState.NOT_INSTALLED:
-            return await self.async_step_install_zigbee_flasher_addon()
-
-        if addon_info.state == AddonState.NOT_RUNNING:
-            return await self.async_step_run_zigbee_flasher_addon()
-
-        # If the addon is already installed and running, fail
-        return self.async_abort(
-            reason="addon_already_running",
-            description_placeholders={
-                **self._get_translation_placeholders(),
-                "addon_name": fw_flasher_manager.addon_name,
-            },
-        )
-
-    async def async_step_install_zigbee_flasher_addon(
-        self, user_input: dict[str, Any] | None = None
-    ) -> ConfigFlowResult:
-        """Show progress dialog for installing the Zigbee flasher addon."""
-        return await self._install_addon(
-            get_zigbee_flasher_addon_manager(self.hass),
-            "install_zigbee_flasher_addon",
-            "run_zigbee_flasher_addon",
-        )
-
-    async def _install_addon(
+    async def _install_firmware_step(
         self,
-        addon_manager: silabs_multiprotocol_addon.WaitingAddonManager,
+        fw_update_url: str,
+        fw_type: str,
+        firmware_name: str,
+        expected_installed_firmware_type: ApplicationType,
         step_id: str,
         next_step_id: str,
     ) -> ConfigFlowResult:
-        """Show progress dialog for installing an addon."""
-        addon_info = await self._async_get_addon_info(addon_manager)
+        assert self._device is not None
 
-        _LOGGER.debug("Flasher addon state: %s", addon_info)
+        if not self.firmware_install_task:
+            session = async_get_clientsession(self.hass)
+            client = FirmwareUpdateClient(fw_update_url, session)
+            manifest = await client.async_update_data()
 
-        if not self.addon_install_task:
-            self.addon_install_task = self.hass.async_create_task(
-                addon_manager.async_install_addon_waiting(),
-                "Addon install",
+            fw_meta = next(
+                fw for fw in manifest.firmwares if fw.filename.startswith(fw_type)
```

Comment:
is there ever going to be a firmware that would match another one? For CPU architectures we could have ie `x86` and `x86-64` (don't know the firmwares here)